pco/src/histograms.rs

```diff
@@ -101,18 +94,22 @@ impl<L: Latent> HistogramBuilder<L> {
       return;
     }
 
-    let tight_ub = match upper {
-      Bound::Loose(_) => slice_max(latents),
-      Bound::Tight(upper) => upper,
-    };
-
     if let Some(bin) = self.incomplete_bin.as_mut() {
-      bin.upper = tight_ub;
+      bin.upper = match upper {
+        Bound::Loose(_) => slice_max(latents),
+        Bound::Tight(upper) => upper,
+      };
       bin.count += latents.len();
     } else {
-      let tight_lb = match lower {
-        Bound::Loose(_) => slice_min(latents),
-        Bound::Tight(lower) => lower,
+      let (tight_lb, tight_ub) = match lower {
```

Comment:
Can you make this a single match on (lower, upper)?

Reply:
Oh, you can match on tuples? Sweet!

Do you have a insights on how often each match arm might be hit so we can order them accordingly?

Reply:
I don't think it matters 😅 This part isn't super hot, and branch prediction will handle it.

But to satisfy your curiosity: we're almost always doing quicksort here, typically pivoting by x from [a, b) into [a, x) and [x, b). So most often it's tight LB, loose UB.

Reply:
Ok, I'll just leave them in alphabetical order then.